api/src/main/java/org/apache/iceberg/SnapshotRef.java

```diff
@@ -139,7 +146,7 @@ public Builder maxRefAgeMs(Long value) {
     }
 
     public SnapshotRef build() {
-      return new SnapshotRef(snapshotId, type, minSnapshotsToKeep, maxSnapshotAgeMs, maxRefAgeMs);
+      return new SnapshotRef(snapshotId, type, minSnapshotsToKeep, maxSnapshotAgeMs, maxRefAgeMs, System.currentTimeMillis());
```

Comment:
I am doubting this is the right way to set a timestamp. We'll need to make sure this is accurate at the time we commit the metadata operation.